app/assets/javascripts/discourse/controllers/preferences/second-factor.js.es6

```diff
@@ -0,0 +1,66 @@
+import { default as computed } from 'ember-addons/ember-computed-decorators';
+import DiscourseURL from 'discourse/lib/url';
+import { userPath } from 'discourse/lib/url';
+import { popupAjaxError } from 'discourse/lib/ajax-error';
+
+export default Ember.Controller.extend({
+
+  loading: false,
+  password: null,
+  second_factor_image: null,
+  second_factor_key: null,
+  show_second_factor_key: false,
+
+  errorMessage: null,
+  newUsername: null,
+
+  @computed('second_factor_image','second_factor_key')
+  loaded(second_factor_image, second_factor_key) {
+    return second_factor_image && second_factor_key;
+  },
+
+  @computed('loading')
+  submitButtonText(loading) {
+    if (loading) return "Loading";
+    return "Submit";
+  },
+
+  actions: {
+    confirmPassword() {
+      if(!this.get('password')) {
+        return;
+      }
+      this.set('loading', true);
+      this.get('content').loadSecondFactorCodes(this.get('password')).then((resp) => {
+        if(resp.error) {
+          this.set('errorMessage',resp.error);
+          return;
+        }
+        this.set('errorMessage',null);
+        this.set('second_factor_key', resp.key);
+        this.set('second_factor_image', resp.qr);
+      })
+        .catch(popupAjaxError)
```

Comment:
This indentation is odd here. I'd prefer if `.catch` was on the previous line following the paren.